assets/src/stories-editor/components/block-navigation/item.js

```diff
@@ -44,28 +43,47 @@ const parseDropEvent = ( event ) => {
 	return result;
 };
 
-class BlockNavigationItem extends Component {
-	constructor( ...args ) {
-		super( ...args );
+const BlockNavigationItem = ( { block, isSelected, onClick, unMovableBlock } ) => {
+	const { clientId } = block;
 
-		this.state = {
-			isDragging: false,
-		};
-	}
+	const [ isDragging, setIsDragging ] = useState( false );
+
+	const rootClientId = useSelect( ( select ) => {
+		const { getBlockRootClientId } = select( 'core/block-editor' );
+		return getBlockRootClientId( clientId );
+	}, [ clientId ] );
+	const blockOrder = useSelect( ( select ) => {
+		const { getBlockOrder } = select( 'core/block-editor' );
+		return getBlockOrder( rootClientId );
+	}, [ rootClientId ] );
+	const blocks = useSelect( ( select ) => {
+		const { getBlocksByClientId } = select( 'core/block-editor' );
+		return getBlocksByClientId( blockOrder ).map( ( { clientId: id } ) => id ).reverse();
+	}, [ blockOrder ] );
```

Comment:
Why not select all three in a single `useSelect`? I know that technically `rootClientId` might not change just because `clientId` does (it might be a different block on the same page) and then your version is a bit quicker, but the overhead is so tiny, that I don't think it warrants this more complex setup.

I'd just go for:

```
	const {
		rootClientId,
		blockOrder,
		blocks,
	} = useSelect( ( select ) => {
		const { getBlockRootClientId, getBlockOrder, getBlocksByClientId } = select( 'core/block-editor' );
		const rootClientId = getBlockRootClientId( clientId );
		const blockOrder = getBlockOrder( rootClientId );
		const blocks = getBlocksByClientId( blockOrder ).map( ( { clientId: id } ) => id ).reverse();
		return { rootClientId, blockOrder, blocks };
	}, [ clientId ] );
```

Reply:
Sure that also works 👍